From a3bdef07d2bd940ae1973b6dffa4a7ddee619122 Mon Sep 17 00:00:00 2001
From: Jeremy Kerr <jk@ozlabs.org>
Date: Wed, 11 Dec 2019 08:55:07 +0800
Subject: [PATCH 10/10] kexec: improve kexec_file_load error message

This change cleans up the error message we get on file_load failure:

 - indicate that it's a failure,
 - no need to mention the internal function name, just that it's a
   file_load issue; and
 - no need to report the syscall return value, it's guaranteed to be -1.

Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
---
 kexec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kexec.c b/kexec.c
index 04cb0cf..31c2bd5 100644
--- a/kexec.c
+++ b/kexec.c
@@ -823,7 +823,7 @@ static void do_file_load(char *kernel, char *initrd, char *cmdline)
 	debug_printf("cmdline=\"%s\"\n", cmdline);
 	ret = syscall_kexec_file_load(kernel_fd, initrd_fd, cmdline_len, cmdline, flags);
 	if (ret) {
-		fprintf(stderr, "do_file_load: (%d) %s\n", ret, strerror(errno));
+		fprintf(stderr, "file_load failed: %s\n", strerror(errno));
 		exit(1);
 	}
 }
-- 
2.17.1

